Source/WebCore/platform/mediastream/gstreamer/GStreamerVideoEncoder.cpp

```diff
@@ -460,6 +462,23 @@ static void webkit_webrtc_video_encoder_class_init(WebKitWebrtcVideoEncoderClass
             g_object_set(self->priv->inputCapsFilter.get(), "caps", inputCaps.get(), nullptr);
         }, "target-bitrate", setBitrateBitPerSec, "keyframe-max-dist");
 
+    Encoders::registerEncoder(VaapiH264, "vah264lpenc", "h264parse", "video/x-h264", nullptr,
+        [](WebKitWebrtcVideoEncoder* self) {
+            g_object_set(self->priv->parser.get(), "config-interval", 1, nullptr);
+        }, "bitrate", setBitrateKbitPerSec, "key-int-max");
+
+    Encoders::registerEncoder(VaapiH264, "vah264enc", "h264parse", "video/x-h264", nullptr,
+        [](WebKitWebrtcVideoEncoder* self) {
+            gst_util_set_object_arg(G_OBJECT(self->priv->encoder.get()), "rate-control", "cbr");
+            g_object_set(self->priv->parser.get(), "config-interval", 1, nullptr);
+        }, "bitrate", setBitrateKbitPerSec, "key-int-max");
```

Comment:
so, basically, iiuc, it will try to register vah264lpenc and if it doesn't exist, it will register vah264enc. Which kind of make sense, because, so far, only Intel provides low power encoders.

What worries me is that intel's vah264lpenc only provides constant quantizer as rate control, while vah264enc usually provides constant bitrate for rate control, which is more comon in streaming, and the bitrate parameter is actually used.

https://slhck.info/video/2017/03/01/rate-control.html

Still CBQ is reported to work well in streaming but you have to deal with other parametrization.